LevelEditor/SliderWidget.py

```diff
@@ -0,0 +1,197 @@
+from PySide.QtCore import *
+from PySide.QtGui import *
+from numpy import mean
+from collections import OrderedDict
+
+sliders = None
+
+
+class AdjustValues(QWidget):
+    valuesChanged = Signal(list)
+
+    def __init__(self, val_list, min_val, max_val):
+        super(AdjustValues, self).__init__()
+        self.val_list = val_list
+        self.min_val = min_val
+        self.max_val = max_val
+        self.slider_count = len(val_list)
+        self.slides = OrderedDict()
+
+        # Insert Sliders
+        self.wrap = QVBoxLayout()
+
+        for i in range(len(val_list)):
+            row = QHBoxLayout()
+            lab = QLabel(str(val_list[i]))
+            slide = QSlider(Qt.Horizontal)
+
+            # remove button
+            rem_button = QPushButton()
+            rem_button.setText("X")
+            rem_button.clicked.connect(self.remove_level(row))
+
+            # setting slider attributes
+            slide.setRange(min_val, max_val)
+            slide.setTickPosition(QSlider.TicksAbove)
+            slide.sliderMoved.connect(self.change_label(lab, slide))
+            slide.sliderMoved.connect(self.adjust_slides(slide))
+            slide.sliderMoved.connect(self.send_values)
+            slide.setValue(val_list[i])
+
+            # label = l, slider = s, position = p
+
+            d = {"l": lab, "s": slide, "p": val_list[i]}
+
+            self.slides[slide] = d
```

Comment:
So, as far as I can tell, you don't really need to hold onto a reference to the label. Instead of using the `sliderMoved` signal, you can use the `valueChanged` signal (which should emit when you use `setValue` on a slider) to update the labels. Since you also don't really need to store the `p` value (which you can just retrieve from the slider), you can just make slides be a `list` of the `QSlider`s.